extensions/infinispan-client/deployment/src/main/java/io/quarkus/infinispan/client/deployment/InfinispanClientProcessor.java

```diff
@@ -207,28 +284,138 @@ InfinispanPropertiesBuildItem setup(ApplicationArchivesBuildItem applicationArch
 
-        return new InfinispanPropertiesBuildItem(properties);
+        return new InfinispanPropertiesBuildItem(propertiesMap);
+    }
+
+    @BuildStep
+    @Record(ExecutionTime.STATIC_INIT)
+    BeanContainerListenerBuildItem build(InfinispanRecorder recorder, InfinispanPropertiesBuildItem builderBuildItem) {
+        Map<String, Properties> propertiesMap = builderBuildItem.getProperties();
+
+        addMaxEntries(DEFAULT_INFINISPAN_CLIENT_NAME,
+                infinispanClientsBuildTimeConfig.defaultInfinispanClient, propertiesMap.get(DEFAULT_INFINISPAN_CLIENT_NAME));
+        for (Map.Entry<String, InfinispanClientBuildTimeConfig> config : infinispanClientsBuildTimeConfig.namedInfinispanClients
+                .entrySet()) {
+            addMaxEntries(config.getKey(), config.getValue(), propertiesMap.get(config.getKey()));
+        }
+
+        // This is necessary to be done for Protostream Marshaller init in native
+        return new BeanContainerListenerBuildItem(recorder.configureInfinispan(propertiesMap));
+    }
+
+    /**
+     * Reads all the contents of the file as a single string using default charset
+     *
+     * @param fileName file on class path to read contents of
+     * @return string containing the contents of the file
+     */
+    private static String getContents(String fileName) {
+        InputStream stream = InfinispanClientProducer.class.getResourceAsStream(fileName);
+        return getContents(stream);
+    }
+
+    /**
+     * Reads all the contents of the input stream as a single string using default charset
+     *
+     * @param stream to read contents of
+     * @return string containing the contents of the file
+     */
+    private static String getContents(InputStream stream) {
+        try (Scanner scanner = new Scanner(stream, "UTF-8")) {
+            return scanner.useDelimiter("\\A").next();
+        }
+    }
+
+    private Set<String> infinispanClientNames(CombinedIndexBuildItem indexBuildItem,
+            BuildProducer<InfinispanClientNameBuildItem> infinispanClientNames) {
+        Set<String> clientNames = new HashSet<>();
+        IndexView indexView = indexBuildItem.getIndex();
+        // adds to clientNames all the client names scanned from @InfinispanClientName annotation
+        addInfinispanClientNameValues(indexView, clientNames);
+        // dev mode client name for default
+        if (infinispanClientsBuildTimeConfig.defaultInfinispanClient.devService.devservices.enabled) {
+            clientNames.add(DEFAULT_INFINISPAN_CLIENT_NAME);
+        }
+
+        for (String clientName : clientNames) {
+            // Produce a client name for each client (produces later RemoteCacheManager and CounterManager instances)
+            infinispanClientNames.produce(new InfinispanClientNameBuildItem(clientName));
+        }
+        return clientNames;
+    }
+
+    private void addInfinispanClientNameValues(IndexView indexView, Set<String> clientNames) {
```

Comment:
Nitpicker note: this method is basically just one for cycle and only used from one place; I wouldn't place it in a separate method.

Reply:
ok